src/test/unit/link_quality_unittest.cc

```diff
@@ -0,0 +1,545 @@
+ */
+TEST(LQTest, TestElement_LQ_SOURCE_NONE_SAMPLES)
+{
+    // given
+
+
+    linkQualitySource = LQ_SOURCE_NONE;
+
+    osdConfigMutable()->item_pos[OSD_LINK_QUALITY] = OSD_POS(8, 1) | OSD_PROFILE_1_FLAG;
+    osdConfigMutable()->link_quality_alarm = 0;
+
+    osdAnalyzeActiveElements();
+
+    // when samples populated 100%
+    for (int x = 0; x < LINK_QUALITY_SAMPLE_COUNT; x++) {
+        setLinkQualityDirect(updateLinkQualitySamples(LINK_QUALITY_MAX_VALUE));
+    }
+
+
+    displayClearScreen(&testDisplayPort);
+    osdRefresh(simulationTime);
+
+    // then
+    displayPortTestBufferSubstring(8, 1, "9");
+
+    // when  updateLinkQualitySamples used   50% rounds to 4
+    for (int x = 0; x < LINK_QUALITY_SAMPLE_COUNT; x++) {
+        setLinkQualityDirect(updateLinkQualitySamples(LINK_QUALITY_MAX_VALUE));
+        setLinkQualityDirect(updateLinkQualitySamples(0));
+    }
+
+    displayClearScreen(&testDisplayPort);
+    osdRefresh(simulationTime);
+
+    // then
+    displayPortTestBufferSubstring(8, 1, "4");
+
+}
+/*
+ * Tests the Tests the OSD_LINK_QUALITY element values  default LQ_SOURCE_NONE
+ */
+TEST(LQTest, TestElement_LQ_SOURCE_NONE_VALUES)
+{
+    // given
+
+
+    linkQualitySource = LQ_SOURCE_NONE;
+
+    osdConfigMutable()->item_pos[OSD_LINK_QUALITY] = OSD_POS(8, 1) | OSD_PROFILE_1_FLAG;
+    osdConfigMutable()->link_quality_alarm = 0;
+
+    osdAnalyzeActiveElements();
+    // when LINK_QUALITY_MAX_VALUE to 1 by 10%
+    uint16_t testscale = 0;
+    for (int testdigit = 10; testdigit > 0; testdigit--) {
+        testscale = testdigit * 102.3;
+        setLinkQualityDirect(testscale);
+        displayClearScreen(&testDisplayPort);
+        osdRefresh(simulationTime);
+#ifdef DEBUG_OSD
+        printf("%d %d\n",testscale, testdigit);
+        displayPortTestPrint();
+#endif
+        // then
+        if (testdigit >= 10){
+            displayPortTestBufferSubstring(7, 1," 9");
+        }else{
+            displayPortTestBufferSubstring(7, 1," %1d", testdigit - 1);
+        }
+    }
+}
+/*
+ * Tests the OSD_LINK_QUALITY element LQ RX_PROTOCOL_CRSF.
+ */
+TEST(LQTest, TestElementLQ_PROTOCOL_CRSF_VALUES)
+{
+    // given
+    linkQualitySource = LQ_SOURCE_RX_PROTOCOL_CRSF;
+
+    osdConfigMutable()->item_pos[OSD_LINK_QUALITY] = OSD_POS(8, 1) | OSD_PROFILE_1_FLAG;
+    osdConfigMutable()->link_quality_alarm = 0;
+
+    osdAnalyzeActiveElements();
+
+    displayClearScreen(&testDisplayPort);
+    osdRefresh(simulationTime);
+
+    // crsf setLinkQualityDirect 0-300;
+
+    for (uint16_t x = 0; x <= 300; x++) {
+        // when x scaled
+        setLinkQualityDirect(scaleCrsfLq(x));
+        // then rxGetLinkQuality Osd should be x
+        displayClearScreen(&testDisplayPort);
+        osdRefresh(simulationTime);
+        displayPortTestBufferSubstring(8, 1,"%3d", x);
+
+    }
+}
+/*
+ * Tests the LQ Alarms
+ *
+*/
+TEST(LQTest, TestLQAlarms_DEFUALT)
```

Comment:
Typo.

Reply:
i hope you mean TestLQAlarms_DEFUALT to TestLQAlarms_LQ_SOURCE_NONE
or just TestLQAlarms ?
... or some where else and it has nothing to do with TestLQAlarms_DEFUALT

Reply:
dyslexia gets me every time ...
 well is should be TestLQAlarms  or TestLQAlarms_LQ_SOURCE_NONE regardless